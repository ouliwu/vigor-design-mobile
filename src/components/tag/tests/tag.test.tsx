import React from 'react'
import { render, testA11y } from 'testing'
import Tag from '..'

describe('Tag', () => {
  test('a11y', async () => {
    await testA11y(<Tag>tag</Tag>)
  })

  test('renders with color', async () => {
    const { container } = render(
      <>
        <Tag color='#primary'>primary</Tag>
        <Tag color='#2db7f5'>#2db7f5</Tag>
      </>
    )

    expect(container).toMatchSnapshot()
  })

  test('renders with fill', async () => {
    const { container } = render(
      <>
        <Tag fill='outline'>outline</Tag>
        <Tag fill='solid'>solid</Tag>
      </>
    )

    expect(container).toMatchSnapshot()
  })
})
