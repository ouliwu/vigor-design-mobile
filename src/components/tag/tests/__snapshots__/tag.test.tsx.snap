// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag renders with color 1`] = `
<div>
  <span
    class="adm-tag"
    style="--border-color: #primary; --text-color: #ffffff; --background-color: #primary;"
  >
    primary
  </span>
  <span
    class="adm-tag"
    style="--border-color: #2db7f5; --text-color: #ffffff; --background-color: #2db7f5;"
  >
    #2db7f5
  </span>
</div>
`;

exports[`Tag renders with fill 1`] = `
<div>
  <span
    class="adm-tag"
    style="--border-color: #666666; --text-color: #666666; --background-color: transparent;"
  >
    outline
  </span>
  <span
    class="adm-tag"
    style="--border-color: #666666; --text-color: #ffffff; --background-color: #666666;"
  >
    solid
  </span>
</div>
`;
